clients/githubrepo/graphql.go

```diff
@@ -161,9 +161,9 @@ type checkRunsGraphqlData struct {
 							}
 						} `graphql:"associatedPullRequests(first: $pullRequestsToAnalyze)"`
 					}
-				} `graphql:"history(first: $commitsToAnalyze)"`
+				} `graphql:"history(first:1)"`
 			} `graphql:"... on Commit"`
-		} `graphql:"object(expression: $commitExpression)"`
+		} `graphql:"object(oid: $commitExpression)"`
```

Comment:
The CheckSuiteConnection and CheckSuiteEdge objects (https://docs.github.com/en/graphql/reference/objects#checksuiteconnection) return a pagination cursor. Could we use that? It would avoid requiring `handler.setup` to be called before `handler.setupCheckRuns`, can fetch multiple checkRuns in a single call saving extra GraphQL calls and we can continue supporting `expression` instead of `oid`. What do you think?

Reply:
i tried to get that working, but for some reason it was still throwing the misses. () Is it possible to have the query i wrote before only trigger when commit depth goes above 100? 
as far as i know cron is running with default (30) commits and that could stay the same. 

also the cache misses only trigger when going above 100 which is a very specific case and would only run when running locally using the users token. seems to work great even with 1000 commits. let me know what you think about this approach

Reply:
> i tried to get that working, but for some reason it was still throwing the misses.

Hmm, interesting. Do you mind sharing the code for this so that I can take a look and see if I can debug more into it?

> Is it possible to have the query i wrote before only trigger when commit depth goes above 100?

I don't feel comfortable leaving the code in that state. I'd rather we explicitly call out cache misses when there are >100 commits instead of silently switching over to making one HTTP call per commit. If we can only handle the case upto 100 commits elegantly, then maybe that's a good enough for now and we can re-visit >100 commit usecase in a future PR.